### DISCUSSION TITLE
Question about re-authentication process in nat-nat ipsec based on swanctl

### DISCUSSION THREAD

**Discussion by alex-airspan** [strongswan/strongswan#1928] [Q&A]
I have IPSEC tunnel UP and traffic running. According to below swanctl --list-sas. 

I understand that gw-gw: **#** incremented every reauth time ( with same cert ) and net-net: # - every rekey ? It Correct?
It looks working when I dont change certificate. 

Now I want only change between re-authentications the certificate ( because my older certificate expired ). After next re-authentication 
I see that gw-gw: **#** back to 1 and any issue with traffic. I see ESP packets received, but counters not really working and traffic over this tunnel very very not stable. 

Question - if possible change certificates without stop the tunnel ?

gw-gw: #**3**, ESTABLISHED, IKEv2, 6946200d2722c2f0_i* 01913b45a7e34816_r
  local  'C=US, O=Networks Inc., OU=Production, CN=00B00B005AFC' @ 172.27.128.193[500] [172.27.124.
  remote 'CN=ilstrongsecgw1, C=US, O=Networks Inc., OU=SGW' @ 172.27.123.10[500]
  AES_CBC-256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
  established 4390s ago, reauth in 6389s
  net-net: #**211**, reqid 1, INSTALLED, TUNNEL, ESP:AES_CBC-256/HMAC_SHA2_256_128
    installed 17s ago, rekeying in 13251s, expires in 15867s
    in  c4254688 (0x0000002a), 416345326 bytes, 311871 packets,     0s ago
    out cb00561c (0x0000002a), 515177 bytes,   571 packets,     0s ago
    local  172.27.124.10/32
    remote 0.0.0.0/0

Thanks,
Alex

**Comment by tobiasbrunner** [comment]
> I understand that gw-gw: **#** incremented every reauth time ( with same cert ) and net-net: # - every rekey ?

They are the unique IDs of each SA, they change with every rekeying (or reauthentication if you use that instead of or in addition to rekeying for IKE).

> Now I want only change between re-authentications the certificate ( because my older certificate expired ). After next re-authentication
> I see that gw-gw: **#** back to 1 and any issue with traffic.

Sounds like you restarted the daemon. The counters are obviously reset then. Not sure why that would cause traffic issues, though.

> Question - if possible change certificates without stop the tunnel ?

Depends on how you reference the certificate. If you explicitly configure it in the connection, then no, as the next reauthentication will use the same certificate (you have to reload the connection, then terminate and re-initiate the connection manually to pick up the new certificate). However, if you just configure the identity in the connection and load the certificate directly (e.g. by placing it in the `swanctl/x509` directory), then the new certificate should be picked up.

**Comment by alex-airspan** [comment]
Hi Tobias,

You correct - I found that daemon restarted. After fixing, I manual copy the certs/key to swanctl folders - re-authentication success with new certs and no issue with traffic. 

One additional question - If possible "gracefully" restart the IPSEC tunnel ? 

Thanks,
Alex

**Reply by tobiasbrunner** [reply]
> If possible "gracefully" restart the IPSEC tunnel ?

What do you mean?

**Comment by alex-airspan** [comment]
I mean if possible send from client to server any message that tunnel going to down, because if I "close" the client, Server will send dpd and only after some time will delete the tunnel configuration

Thanks,
Alex

**Reply by tobiasbrunner** [reply]
> because if I "close" the client

What do you mean with "close"? You don't terminate the connection or the daemon so a DELETE is sent?

> Server will send dpd and only after some time will delete the tunnel configuration

Sure, but that's usually not a problem. Depending on how mobile the clients are, you don't want to keep that DPD timeout too short anyway so a client can be without connectivity for a while.
